qualtran/bloqs/gf_arithmetic/gf2_addition.py

```diff
@@ -0,0 +1,99 @@
+#  Copyright 2024 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from functools import cached_property
+from typing import Dict, Set, TYPE_CHECKING, Union
+
+import attrs
+
+from qualtran import Bloq, bloq_example, BloqDocSpec, DecomposeTypeError, QGF, Register, Signature
+from qualtran.bloqs.basic_gates import CNOT
+from qualtran.symbolics import is_symbolic, SymbolicInt
+
+if TYPE_CHECKING:
+    from qualtran import BloqBuilder, Soquet
+    from qualtran.resource_counting import BloqCountDictT, BloqCountT, SympySymbolAllocator
+    from qualtran.simulation.classical_sim import ClassicalValT
+
+
+@attrs.frozen
+class GF2Addition(Bloq):
+    r"""In place addition over GF($2^m$).
+
+    The bloq implements in place addition of two quantum registers storing elements
+    from GF($2^m$). Addition in GF($2^m$) simply reduces to a component wise XOR, which
+    can be implemented via CNOT gates. The addition is performed in-place such that
+
+        $$
+        |x\rangle |y\rangle \rightarrow |x\rangle |x + y\rangle
```

Comment:
markdown treats indented blocks like this as verbatim / code. Can you please unindent?

Reply:
done